### PR TITLE
ConsoleCalculator.kraven88

### PR DESCRIPTION
**Pull request by kraven88**
### Review version 1.0.1

All additional challenges completed.

Since the purpose of this project was to follow a written documentation based on MS tutorial, 
the Calculator.DoOperation() method remains in mostly unchanged form since the beginning.
I've only changed how to app handles saving equations to a JSON file. The rest is added on top
of the original project.

### CODE REVIEW

File: CalculatorLibrary/Calculator.cs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+namespace CalculatorLibrary;
+
+public class Calculator
+{
+    private List<Equation> equations;
+    private string filePath;
+    public int counter = 0;    // Number of times the Calculator has been used
+
+    public Calculator(List<Equation> equations, string filePath)
```
Comment (TheCSharpAcademy) [inline CalculatorLibrary/Calculator.cs:11]: If you change the arguments names you can get rid of the ugly "this" 😁
image

**Review by TheCSharpAcademy — CHANGES_REQUESTED** [review]

File: ConsoleCalculator.kraven88/MainMenu.cs [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+            if (Console.ReadLine().ToLower() == "n")
+                nextEquasion = false;
+
+            Console.WriteLine("\n"); // Friendly linespacing.
+        }
+        
+        return true;
+    }
+
+    private bool PreviousEquations(Calculator calculator)
```
[the remark this answers is not in the document]
Comment (kraven88) [reply ConsoleCalculator.kraven88/MainMenu.cs:132]: It is called from SelectionScreen (line 31)

File: CalculatorLibrary/Calculator.cs (hunk 1 above)
Comment (kraven88) [reply CalculatorLibrary/Calculator.cs:11]: I saw 2 ways of doing it. One was with underscore and the other with "this". I preferred the latter and it didn't bothered me at all. I honestly consider things like "_equations" even uglier :)

File: CalculatorLibrary/Calculator.cs [hunk 3]
```diff
@@ -0,0 +1,81 @@
+﻿using CalculatorLibrary.Models;
+
+namespace CalculatorLibrary;
+
+public class Calculator
+{
+    private List<Equation> equations;
+    private string filePath;
+    public int counter = 0;    // Number of times the Calculator has been used
```
[the remark this answers is not in the document]
Comment (kraven88) [reply CalculatorLibrary/Calculator.cs:9]: I think this is a remnant of some comments I made when making stuff, so I don't forget to actually do them :)
Will remove.

File: ConsoleCalculator.kraven88/MainMenu.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (TheCSharpAcademy) [reply ConsoleCalculator.kraven88/MainMenu.cs:132]: Sorry, I mean the "calculator" parameter isn't being used: 
image

File: CalculatorLibrary/Calculator.cs (hunk 1 above)
Comment (TheCSharpAcademy) [reply CalculatorLibrary/Calculator.cs:11]: @kraven88 yeah at the end of the day it's your personal preference. In enterprise you'll follow whatever code standard you company has 🤓😁

**Pull request merged**